receiver/zookeeperreceiver/scraper_test.go

```diff
@@ -254,50 +219,27 @@ func TestZookeeperMetricsScraperScrape(t *testing.T) {
 				require.Equal(t, log.level, observedLogs.All()[i].Level)
 			}
 
-			if tt.wantErr {
-				require.Error(t, err)
-				require.Equal(t, pdata.NewMetrics(), got)
+			if tt.expectedNumResourceMetrics == 0 {
+				if tt.wantErr {
+					require.Error(t, err)
+					require.Equal(t, pdata.NewMetrics(), got)
+				}
 
 				require.NoError(t, z.shutdown(ctx))
 				return
 			}
 
-			require.Equal(t, tt.expectedNumResourceMetrics, got.ResourceMetrics().Len())
-			for i := 0; i < tt.expectedNumResourceMetrics; i++ {
-				resource := got.ResourceMetrics().At(i).Resource()
-				require.Equal(t, len(tt.expectedResourceAttributes), resource.Attributes().Len())
-				resource.Attributes().Range(func(k string, v pdata.AttributeValue) bool {
-					require.Equal(t, tt.expectedResourceAttributes[k], v.StringVal())
-					return true
-				})
-
-				ilms := got.ResourceMetrics().At(0).InstrumentationLibraryMetrics()
-				require.Equal(t, 1, ilms.Len())
-				metrics := ilms.At(0).Metrics()
-
-				require.Equal(t, len(tt.expectedMetrics), metrics.Len())
-
-				for i := 0; i < metrics.Len(); i++ {
-					assertMetricValid(t, metrics.At(i), tt.expectedMetrics)
-				}
-			}
+			expectedFile := filepath.Join("testdata", "scraper", fmt.Sprintf("%s.json", tt.expectedMetricsFilename))
+			expectedMetrics, err := scrapertest.ReadExpected(expectedFile)
+			require.NoError(t, err)
+			eMetricSlice := expectedMetrics.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics()
+			aMetricSlice := got.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics()
 
-			require.NoError(t, z.shutdown(ctx))
+			require.NoError(t, scrapertest.CompareMetricSlices(eMetricSlice, aMetricSlice))
```

Comment:
I'm wondering if we should have `scrapertest.CompareMetrics(expected, got pdata.Metrics)` helper as well?

Reply:
Yes, I think we should. Haven't needed it yet but will soon. I'll plan on adding this.